lms/djangoapps/instructor_task/models.py

```diff
@@ -321,19 +321,30 @@ def store_rows(self, course_id, filename, rows):
 
         self.store(course_id, filename, output_buffer)
 
+    def delete_file(self, course_id, filename):
+        """
+        Given the `course_id` and `filename` for the report, this method deletes the report
+        """
+        key = self.key_for(course_id, filename)
+        self.bucket.delete_key(key)
+
     def links_for(self, course_id):
         """
         For a given `course_id`, return a list of `(filename, url)` tuples. `url`
         can be plugged straight into an href
         """
         course_dir = self.key_for(course_id, '')
-        return sorted(
+        sorted_link_data = sorted(
             [
-                (key.key.split("/")[-1], key.generate_url(expires_in=300))
+                (key.key.split("/")[-1].split("_")[-1].split('.csv')[0], key.key.split("/")[-1], key.generate_url(expires_in=300))
```

Comment:
i would add some documentation or split this up because right now it's difficult to read